### PR TITLE
Check if we actually got a cache path from nuget

### PR DESCRIPTION
**Pull request by matthewfcarlson**

### CODE REVIEW

File: edk2toolext/environment/extdeptypes/nuget_dependency.py [hunk 1]
```diff
@@ -105,13 +105,16 @@ def _fetch_from_cache(self, package_name):
                 return_buffer.seek(0)
                 return_string = return_buffer.read()
                 NugetDependency.global_cache_path = return_string.strip().strip("global-packages: ")
-
+        
+        if NugetDependency.global_cache_path is None:
+            logging.info("Could not determine Nuget global packages cache location")
```
Comment (spbrogan) [inline edk2toolext/environment/extdeptypes/nuget_dependency.py:110]: should we add return_string into the log so we can evaluate the issue

Comment (matthewfcarlson) [reply edk2toolext/environment/extdeptypes/nuget_dependency.py:110]: In theory it should already be in the ci buildlog

File: edk2toolext/environment/extdeptypes/nuget_dependency.py [hunk 2]
```diff
@@ -106,12 +106,15 @@ def _fetch_from_cache(self, package_name):
...
+            logging.info("Check your mono installation")
+            return False
         #
         # If the path couldn't be found, we can't do anything else.
         #
         if not os.path.isdir(NugetDependency.global_cache_path):
-            logging.info(
-                "Could not determine Nuget global packages cache location.")
+            logging.info("Could not determine Nuget global packages cache location.")
```
Comment (spbrogan) [inline edk2toolext/environment/extdeptypes/nuget_dependency.py:117]: this message is the same as the above and seems wrong

Comment (spbrogan) [reply edk2toolext/environment/extdeptypes/nuget_dependency.py:117]: seems like it should be...global cache path is not a directory

Comment (matthewfcarlson) [reply edk2toolext/environment/extdeptypes/nuget_dependency.py:116]: Yeah- fixed

File: edk2toolext/environment/extdeptypes/nuget_dependency.py [hunk 3]
```diff
@@ -105,13 +105,16 @@ def _fetch_from_cache(self, package_name):
                 return_buffer.seek(0)
                 return_string = return_buffer.read()
                 NugetDependency.global_cache_path = return_string.strip().strip("global-packages: ")
-
+        
+        if NugetDependency.global_cache_path is None:
+            logging.info("Could not determine Nuget global packages cache location")
+            logging.info("Check your mono installation")
```
Comment (spbrogan) [inline edk2toolext/environment/extdeptypes/nuget_dependency.py]: dump this

**Comment by matthewfcarlson** [comment]
Also moved global_cache_path local

**Pull request merged**